### ISSUE TITLE
Wrong jump on failed assumes

### ISSUE THREAD

**Issue by ThomasHaas** [hernanponcedeleon/Dat3M#51]
A failed assume inside a function call causes a jump to the end of the function call instead of end of the thread (it has no abort semantics).

Example
```
 210:    if((! (! 8:$nextEqNull == True))); then goto END_OF_8: // Compiled Assume
...
 246: END_OF_8: // End of function
 247:    === Returning from dequeue ===
```

**Comment by hernanponcedeleon** [comment]
I pushed the fix to `address_simplify` (713770f52d3b2c627de7727ad2b5ca0f45f67da5) because for testing on benchmarks like the one you posted I need some of the changes that are only in the branch.
The jumps looks correct to me now, but can you please double check?
I'll keep this issue open anyway until the changes are merged into `development`.

**Comment by hernanponcedeleon** [comment]
Merged to `development`
